src/test/lib/ApiTest.cpp

```diff
@@ -1152,6 +1238,69 @@ DummyStreamCallback(
     return QUIC_STATUS_SUCCESS;
 }
 
+struct CloseFromCallbackContext {
+    uint16_t CloseCount;
+    uint16_t CurrentCount;
+
+    static QUIC_STATUS StreamCallback(_In_ MsQuicStream*, _In_opt_ void*, _Inout_ QUIC_STREAM_EVENT*) {
+        return QUIC_STATUS_SUCCESS;
+    }
+
+    static QUIC_STATUS Callback(_In_ MsQuicConnection* Conn, _In_opt_ void* Context, _Inout_ QUIC_CONNECTION_EVENT* Event) {
+        auto Ctx = (CloseFromCallbackContext*)Context;
+
+        if (Event->Type == QUIC_CONNECTION_EVENT_PEER_STREAM_STARTED) {
+            new(std::nothrow) MsQuicStream(Event->PEER_STREAM_STARTED.Stream, CleanUpAutoDelete, StreamCallback, nullptr);
+        }
+        if (Ctx->CloseCount == Ctx->CurrentCount++) {
```

Comment:
You should use `InterlockedIncrement16` here instead, because this can be called from separate threads.